core/templates/pages/exploration-editor-page/services/exploration-rights-backend-api.service.spec.ts

```diff
@@ -0,0 +1,222 @@
+
+describe('Exploration Rights Backend Api Service', () => {
+  let service: ExplorationRightsBackendApiService;
+  let httpTestingController: HttpTestingController;
+  let successHandler = jasmine.createSpy('success');
+  let failHandler = jasmine.createSpy('fail');
+
+  beforeEach(waitForAsync(() => {
+    TestBed.configureTestingModule({
+      imports: [HttpClientTestingModule],
+      providers: [ExplorationRightsBackendApiService]
+    });
+    httpTestingController = TestBed.inject(HttpTestingController);
+    service = TestBed.inject(ExplorationRightsBackendApiService);
+  }));
+
+  afterEach(() => {
+    httpTestingController.verify();
+  });
+
+  it('should put requestData when makeCommunityOwnedPutData called', fakeAsync(
+    () => {
+      let requestData = {
+        version: 3,
+        make_communityOwned: true
+      };
+
+      service.makeCommunityOwnedPutData(
+        'oppia12345', requestData.version, requestData.make_communityOwned
+      ).then(successHandler, failHandler);
+
+      let req = httpTestingController.expectOne(
+        '/createhandler/rights/oppia12345');
+      expect(req.request.method).toEqual('PUT');
+      expect(req.request.body).toEqual(requestData);
+      req.flush([]);
+
+      flushMicrotasks();
+
+      expect(successHandler).toHaveBeenCalled();
+      expect(failHandler).not.toHaveBeenCalled();
+    })
+  );
+
+  it('should put requestData when saveRoleChangesPutData called', fakeAsync(
+    () => {
+      let requestData = {
+        version: 3,
+        new_member_role: 'editor',
+        new_member_username: 'usernameForEditorRole'
+      };
+
+      service.saveRoleChangesPutData(
+        'oppia12345', requestData.version, requestData.new_member_role,
+        requestData.new_member_username
+      ).then(successHandler, failHandler);
+
+      let req = httpTestingController.expectOne(
+        '/createhandler/rights/oppia12345');
+      expect(req.request.method).toEqual('PUT');
+      expect(req.request.body).toEqual(requestData);
+      req.flush([]);
+
+      flushMicrotasks();
+
+      expect(successHandler).toHaveBeenCalled();
+      expect(failHandler).not.toHaveBeenCalled();
+    })
+  );
+
+  it('should put requestData when setViewabilityPutData called', fakeAsync(
+    () => {
+      let requestData = {
+        version: 3,
+        viewableIfPrivate: true
+      };
+
+      service.setViewabilityPutData(
+        'oppia12345', requestData.version, requestData.viewableIfPrivate
+      ).then(successHandler, failHandler);
+
+      let req = httpTestingController.expectOne(
+        '/createhandler/rights/oppia12345');
+      expect(req.request.method).toEqual('PUT');
+      expect(req.request.body).toEqual(requestData);
+      req.flush([]);
+
+      flushMicrotasks();
+
+      expect(successHandler).toHaveBeenCalled();
+      expect(failHandler).not.toHaveBeenCalled();
+    })
+  );
+
+  it('should put requestData when publishPutData called', fakeAsync(() => {
```

Comment:
ditto comment for all tests

Reply:
tried to make the test name more clear ( Done )